### PR TITLE
Update system error

### PR DESCRIPTION
**Pull request by aashikam**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java [hunk 1]
```diff
@@ -55,33 +55,30 @@
...
-        } else {
-            reason = SystemConstants.ERROR_REASON_PREFIX + UNKNOWN_REASON;
-        }
-        if (details != null) {
-            refData.put("message", details);
+    public static ErrorValue getBallerinaError(String details) {
+        return BallerinaErrors.createError(SystemConstants.SYSTEM_ERROR_CODE, populateFilepathErrorRecord(details));
+    }
+
+    private static MapValue populateFilepathErrorRecord(String message) {
```
Comment (daneshk) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java]: Rename the method to `populateSystemErrorRecord`

Comment (aashikam) [reply stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java]: Resolved.

File: stdlib/system/src/main/ballerina/src/system/system.bal [hunk 2]
```diff
@@ -53,23 +53,23 @@ public function exists(string path) returns boolean = external;
...
 # + return - Returns absolute path value of the created directory or an `error` if failed
-public function createDir(string dir, boolean parentDirs = false) returns string|error = external;
+public function createDir(string dir, boolean parentDirs = false) returns string|Error = external;
 
 # Removes the specified file or directory.
 # If recursive flag is true, Removes the path and any children  # it contains.
 #
 # + path - String value of file/directory path.
 # + recursive - Indicates whether the `remove` should recursively remove all the file inside the given directory.
 # + return - Returns an `error` if failed to remove.
```
Comment (wggihan) [inline stdlib/system/src/main/ballerina/src/system/system.bal]:
```suggestion
# + return - Returns an `Error` if failed to remove.
```

Comment (aashikam) [reply stdlib/system/src/main/ballerina/src/system/system.bal]: Resolved

File: stdlib/system/src/main/ballerina/src/system/system_errors.bal [hunk 3]
```diff
@@ -23,6 +23,20 @@ type Detail record {
...
+
+public type PermissionError error<PERMISSION_ERROR, Detail>;
+
+public const FILE_SYSTEM_ERROR = "{ballerina/system}FileSystemError";
+
+public type FileSystemError error<FILE_SYSTEM_ERROR, Detail>;
+
+public const OPERATION_FAILED_ERROR = "{ballerina/system}OperationFailedError";
+
+public type OperationFailedError error<OPERATION_FAILED_ERROR, Detail>;
```
Comment (daneshk) [inline stdlib/system/src/main/ballerina/src/system/system_errors.bal]: Shall we remove this?

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/Copy.java [hunk 4]
```diff
@@ -64,12 +64,13 @@ public static Object copy(Strand strand, String sourcePath, String destinationPa
...
         if (Files.notExists(srcPath)) {
-            return SystemUtils.getBallerinaError("INVALID_OPERATION: File doesn't exist in path " + sourcePath);
+            return SystemUtils.getBallerinaError(SystemConstants.INVALID_OPERATION_ERROR,
+                    "File doesn't exist in path " + sourcePath);
         }
         try {
             Files.walkFileTree(srcPath, new RecursiveFileCopyVisitor(srcPath, destPath, replaceExisting));
         } catch (IOException ex) {
-            return SystemUtils.getBallerinaError("OPERATION_FAILED", ex);
+            return SystemUtils.getBallerinaError(SystemConstants.OPERATION_FAILED_ERROR, ex);
```
Comment (daneshk) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/Copy.java]: Make this `FILE_SYSTEM_ERROR` error

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/ReadDir.java [hunk 5]
```diff
@@ -79,9 +81,9 @@ public static Object readDir(Strand strand, String path) {
             }).toArray(ObjectValue[]::new);
             return new ArrayValue(results, new BArrayType(fileInfoType));
         } catch (IOException | BallerinaException ex) {
-            return SystemUtils.getBallerinaError("OPERATION_FAILED", ex);
+            return SystemUtils.getBallerinaError(SystemConstants.OPERATION_FAILED_ERROR, ex);
```
Comment (daneshk) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/ReadDir.java]: Make this FILE_SYSTEM_ERROR error

**Pull request merged**